### PR TITLE
HSEARCH-2569 Make the 'delete-by-query' plugin requirement for ES clearer

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-2569

This PR brings two changes:

 1. The documentation is clearer about the fact that the 'delete-by-query' plugin is required when using the "purgeAll" option on the mass indexer, and this part of the documentation is more visible.
 2. Whenever a user tries to purge indexes and a 404 error occurs, the error message now mentions that the 'delete-by-query' plugin is required.

We'll probably want to backport at least the first commit to 5.6.

### CODE REVIEW

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+/**
+ * @author Yoann Rodiere
+ */
+public class DeleteByQueryResultAssessor implements BackendRequestResultAssessor<JestResult> {
+
+	private static final Log LOG = LoggerFactory.make( Log.class );
+
+	private static final int NOT_FOUNT_HTTP_STATUS_CODE = 404;
```
Comment (gsmet) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java]: s/FOUNT/FOUND/

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	private final JestAPIFormatter formatter;
+
+	private final DefaultBackendRequestResultAssessor delegate;
+
+	public DeleteByQueryResultAssessor(JestAPIFormatter formatter) {
+		super();
```
Comment (gsmet) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java]: No need for the `super()` here.

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java (hunk 1 above)
Comment (yrodiere) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java]: Fixed.

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java (hunk 2 above)
Comment (yrodiere) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/DeleteByQueryResultAssessor.java]: I'll remove it, but I seriously don't know what's the problem with this.

**Comment by Sanne** [comment]
Looks nice, and +1 to backport: I see no significant risk.

Minor question: why did you opt for `org.hibernate.search.elasticsearch.impl.JestAPIFormatter` to be a Service?

(I couldn't test & build it yet, still figuring out what's wrong with my machine..)

**Comment by yrodiere** [comment]
> Minor question: why did you opt for org.hibernate.search.elasticsearch.impl.JestAPIFormatter to be a Service?

Because it needs access to `GsonService`. I guess I could have made it a constructor parameter, but users would still have had to request/release an instance of `GsonService` (instead of `JestAPIFormatter` right now), so it wouldn't have been simpler in the end.

**Comment by Sanne** [comment]
merged on master and 5.6

**Pull request merged**